src/PVRIptvData.cpp

```diff
@@ -802,7 +802,12 @@ bool PVRIptvData::LoadPlayList(void)
       channel.iTvgShift         = tmpChannel.iTvgShift;
       channel.bRadio            = tmpChannel.bRadio;
       channel.properties        = tmpChannel.properties;
-      channel.strStreamURL      = strLine;
+      
+      if (!g_userAgent.empty() && channel.properties.find("http-user-agent") == channel.properties.end())
```

Comment:
What about also populating user-agent from the property if it's present?

Reply:
It's not needed because all properties from the stream are taken here:

https://github.com/kodi-pvr/pvr.iptvsimple/blob/51663196bc3a756c6903e1564ef017ab2dbd66bf/src/PVRIptvData.cpp#L804

Reply:
Yes, but nothing is done with the user-agent property. If just stays in the channel properties and is never used as far as I can see.

Reply:
It must be used, if not all the properties mechanism wouldn't work at all. A thing can be tested quickly with a playlist with properties.

Reply:
All the properties mechanism does is make a key value pair available to the backend. It does not necessarily do anything with it. Certainly not a vlc only property.

Reply:
Exactly, could be a key value pair already available from the stream. I can't delete or edit a property from the playlist, since has higher priority. The only thing i can do it's to use the setting when it's not available.

Reply:
But then why not this:

```
auto propPair = channel.properties.find("http-user-agent");
if (propPair != channel.properties.end())
  channel.strStreamURL = AddHeaderToStreamUrl(strLine, "user-agent", propPair->second);
else (!g_userAgent.empty())
  channel.strStreamURL = AddHeaderToStreamUrl(strLine, "user-agent", g_userAgent);
```

Then you replicate the same functionality there is in Matrix.

Reply:
Yeah, meanwhile i could add that unused property of the playlist.

Reply:
I think that is the unused property of the playlist 😉